updater.go

```diff
@@ -0,0 +1,182 @@
+package deploy
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/remind101/deploy/Godeps/_workspace/src/github.com/inconshreveable/go-update"
+	"github.com/remind101/deploy/Godeps/_workspace/src/github.com/octokit/go-octokit/octokit"
+)
+
+const GitHubHost = "github.com"
+
+type Updater struct {
+	Host           string
+	CurrentVersion string
+}
+
+func NewUpdater() *Updater {
+	version := Version
+	return &Updater{
+		Host:           GitHubHost,
+		CurrentVersion: version,
+	}
+}
+
+func (updater *Updater) Update() (err error) {
+	releaseName, version := updater.latestReleaseNameAndVersion()
+
+	if version == "" {
+		fmt.Println("There is no newer version of Deploy available.")
+		return
+	}
+
+	if version == updater.CurrentVersion {
+		fmt.Printf("You're already on the latest version: %s\n", version)
+	} else {
+		err = updater.updateTo(releaseName, version)
+	}
+
+	return
+}
+
+func (updater *Updater) updateTo(releaseName, version string) (err error) {
+	downloadURL := fmt.Sprintf("https://%s/remind101/deploy/releases/download/%s/deploy_%s_%s_%s.zip", updater.Host, releaseName, version, runtime.GOOS, runtime.GOARCH)
+
+	fmt.Printf("Downloading %s...", version)
+	path, err := downloadFile(downloadURL)
+	if err != nil {
+		return
+	}
+
+	exec, err := unzipExecutable(path)
+	if err != nil {
+		return
+	}
+
+	err, _ = update.New().FromFile(exec)
+	if err != nil {
+		fmt.Printf("Update failed: %v\n", err)
+	}
+
+	return
+}
+
+func unzipExecutable(path string) (exec string, err error) {
```

Comment:
Wonder if we should just upload the binary instead of a zip file. Then we can just remove this.